assets/dev/js/editor/components/template-library/component.js

```diff
@@ -214,21 +214,21 @@ export default class Component extends ComponentModalBase {
 				dialog.show();
 			},
 
-			initDialog: function() {
+			initDialog: function( model ) {
 				InsertTemplateHandler.dialog = elementorCommon.dialogsManager.createWidget( 'confirm', {
 					id: 'elementor-insert-template-settings-dialog',
-					headerMessage: __( 'Import Document Settings', 'elementor' ),
-					message: __( 'Do you want to also import the document settings of the template?', 'elementor' ) + '<br>' + __( 'Attention: Importing may override previous settings.', 'elementor' ),
+					headerMessage: __( 'Apply the settings of this %s too?', 'elementor' ).replace( '%s', model.attributes.type ),
```

Comment:
```suggestion
					headerMessage: __( 'Apply the settings of this %s too?', 'elementor' ).replace( '%s', model.attributes.label ),
```

Check if there is better name then 'type'

Reply:
Done